### PR TITLE
Set stable color for semantics debugger

### LINKED ISSUE #156242: [A11y] have a more stable coloring for semantics debugger
### Use case

Currently a lot of internal test uses semantic debugger for scuba tests. This has created a bunch of noise when updating some semantics that involves id changes.

### Proposal

Right now the coloring is created by hashing node id. Which doesn't really affect the functionality of the semantics. We should consider hashing based on the shape of semantics tree.

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/156242

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by nate-thegrate** [comment]
Looks great to me, but I'll leave reviewing up to someone with access to scuba tests.

(FYI, it looks like the PR needs a rebase)

File: packages/flutter/lib/src/widgets/semantics_debugger.dart
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/semantics_debugger.dart:342]: So this is actually deterministic? Would it be possible to use the node's uid instead of logic in `_getColorSeed` to ensure the uniqueness (Is `math.Random` an injective function)?

Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/semantics_debugger.dart:342]: question: so it looks like `_findDepth` finds the distance between this node and the deepest descendant. Why this instead of using the distance to the root? If we use that depth instead, all root nodes will be of the same color and every node at the same "location" in the tree (`(# of ancestors, child index)` pair) will have a consistent color? Is that worth doing?

File: packages/flutter/test/widgets/semantics_debugger_test.dart
Comment (LongCatIsLooong) [inline packages/flutter/test/widgets/semantics_debugger_test.dart:7]: The CI analyzer doesn't like this it seems.

**Comment by LongCatIsLooong** [comment]
LGTM modulo the CI failure. 

That's a lot of golden changes. I skimmed through a few hundred of them, they all seem ok to me. I wonder if we should change hashing to avoid common color blind colors and those too close to #FF000000 so the semantics labels are legible?

**Comment by LongCatIsLooong** [comment]
I think the analyzer script doesn't like the `import 'package:flutter/rendering.dart';` import? We shall see.

**Review by LongCatIsLooong — APPROVED** [review]

**Comment by zanderso** [comment]
Is this PR intended to fix the tree? It landed with the tree closed.

**Comment by chunhtai** [comment]
I used the merge when ready button. I thought that would wait for the tree to open. am I doing it wrong?

also the message said
Merged via the queue into flutter:master with commit

**Comment by zanderso** [comment]
I see. For what you wanted, you should continue to use the `autosubmit` label. The tree being closed doesn't block submitting via the merge queue. Right now, it's just like clicking the 'merge' button from before the merge queue was enabled.

**Comment by chunhtai** [comment]
looks like that button served a different purpose, and the correct workflow will still be using autosubmit. should I revert this pr?

**Comment by chunhtai** [comment]
Reason for revert: merge by mistake

**Pull request merged**
